### DISCUSSION TITLE
How to use your tool?

### DISCUSSION THREAD

**Discussion by ccoVeille** [Yokozuna59/mermaid-lint#125] [Q&A] [answered]
I'm interested in using your tool but I have no idea how

I'm not a JavaScript developer.

Could you explain how to build a CLI tool I can use in terminal?

Thanks

**Comment by Yokozuna59** [comment, marked as answer]
Hi @ccoVeille, Thank you for your interest.

While I was trying to build the linter, it appears that `mermaid` needs a new parser, so I'm currently building it first; see https://github.com/orgs/mermaid-js/discussions/4362#discussioncomment-5905873.
Moreover, the idea of the parser was adopted by the mermaid team and merged into https://github.com/mermaid-js/mermaid inside the `packages/parser` folder.
As of right now, the new parser only supports three diagrams; if that number were higher, I would definitely continue working on this linting tool.

So, the package/tool is not ready to be used yet.

**Reply by ccoVeille** [reply]
Thanks for replying me.

I'm happy to see there is a current discussion about it on directly on mermaid project. That's a good achievement.

Your project is the only one I found by looking for mermaid linting, maybe you could explain about to the current project status. I mean either link to discussion or the others you mentioned in your Readme file.

Or maybe this repository is now useless, and can be deleted. I suggest, but I have no ideas 😅

Once again, thanks for working on a mermaid linter and for replying me so fast.

**Reply by ccoVeille** [reply]
For anyone interested, here is a comment on mermaid project that sum up the project status

https://github.com/mermaid-js/mermaid/pull/4450#issuecomment-1751932646

**Reply by Yokozuna59** [reply]
> Your project is the only one I found by looking for mermaid linting, maybe you could explain about to the current project status. I mean either link to discussion or the others you mentioned in your Readme file.

You're right, I need to explain the project current status in the README file.

> Or maybe this repository is now useless, and can be deleted. I suggest, but I have no ideas 😅

Right now, the content of the repository is kind of useless since the existing development has already been implemented at https://github.com/mermaid-js/mermaid, but I wouldn't say the repository itself is useless; it brought some attention and showed that some people want such tool.

> Once again, thanks for working on a mermaid linter and for replying me so fast.

No worries. It's motivating to hear that you're excited about this tool. If you have any suggestions, ideas, or feedback, please feel free to share.

**Reply by ccoVeille** [reply]
Thanks for your reply

I opened #126 to help you to think about updating the readme file.

Thanks again for your help on improving mermaid with such a wonderful project

**Comment by ccoVeille** [comment]
For anyone interested in a mermaid-linter there is one available on gitlab ci for the code of GitLab

https://gitlab.com/gitlab-org/gitlab/-/blob/master/scripts/lint/check_mermaid.mjs
